### PR TITLE
updateOn parameter for DataSearch

### PR DESCRIPTION
**Pull request by anoopadvaitha**
Currently DataSearch filters data as soon as we start typing (onchange). This becomes challenge in scenarios where the data is fetched from API. 

#### What does this PR do?
The PR provides updateOn parameter which is passed to DataForm so that developer can leverage to either use change or submit. 

#### Where should the reviewer start?
\src\js\components\DataSearch\DataSearch.js

#### What testing has been done on this PR?
Tested when no value is provided for DataSearch and when submit is provided and works in both scenarios. 

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?
#7024 

#### Do the grommet docs need to be updated?
Yes, will be raising the PR on that

#### Should this PR be mentioned in the release notes?
No.

#### Is this change backwards compatible or is it a breaking change?
Yes

### CODE REVIEW

File: src/js/components/DataSearch/__tests__/DataSearch-test.tsx [hunk 1]
```diff
@@ -60,3 +60,17 @@ describe('DataSearch', () => {
...
+
+  const { container } = render(
+    <Grommet>
+      <Data id="test-data" data={data}>
+        <DataSearch updateOn='submit' />
+      </Data>
+    </Grommet>,
+  );
+
+  expect(container.firstChild).toMatchSnapshot();
```
[the remark this answers is not in the document]
Comment (anoopadvaitha) [reply src/js/components/DataSearch/__tests__/DataSearch-test.tsx:75]: I've updated the test case and it is failing, even after providing a filter value and doing submit/ simulating enter. The snapshot still has other values.
But it is working as expected in storybook.
Can you please help me on this ?

[the remark this answers is not in the document]
Comment (anoopadvaitha) [reply src/js/components/DataSearch/__tests__/DataSearch-test.tsx:75]: I've tried userEvent which is timing out.

**Comment by MikeKingdom** [comment]
Just an update on this PR: I think the `updateOn="submit"` is great and I did a few minor tweaks while looking at how to get the jest test to actually submit the search when enter is pressed. But as we've been looking at a few other issues in the Data* components and how they use a Form we decided we want to remove the use of Form for most of these. So my next steps here are to refactor DataSearch to not use form, and I'll make sure the `updateOn="submit"` is still implemented so you can trigger the search via the enter key rather than while typing each character.

**Comment by anoopadvaitha** [comment]
> Just an update on this PR: I think the `updateOn="submit"` is great and I did a few minor tweaks while looking at how to get the jest test to actually submit the search when enter is pressed. But as we've been looking at a few other issues in the Data* components and how they use a Form we decided we want to remove the use of Form for most of these. So my next steps here are to refactor DataSearch to not use form, and I'll make sure the `updateOn="submit"` is still implemented so you can trigger the search via the enter key rather than while typing each character.

That sounds like a plan! 
Thank you!

**Review by MikeKingdom — COMMENTED** [review]
@anoopadvaitha  With the changes to remove Form from DataSearch I created an newer implementation of this feature in #7110

**Comment by MikeKingdom** [comment]
Superseded by #7110

**Pull request closed without merging**